rust/emitter/src/ast_emitter.rs

```diff
@@ -461,6 +465,27 @@ impl AstEmitter {
         Ok(())
     }
 
+    fn emit_assignment_expression(
+        &mut self,
+        binding: &AssignmentTarget,
+        expression: &Expression
+    ) -> Result<(), EmitError> {
+        match binding {
+            AssignmentTarget::SimpleAssignmentTarget(
+                SimpleAssignmentTarget::AssignmentTargetIdentifier(AssignmentTargetIdentifier {
+                name, ..
+            })) => {
+                self.emit.bind_g_name(name.value);
+                self.emit_expression(expression)?;
+                self.emit.set_g_name(name.value);
+                return Ok(());
+            },
+            _ => {}
+        }
+
+        return Err(EmitError::NotImplemented("TODO: AssignmentExpression"));
```

Comment:
:+1: As far as I can tell this bytecode is correct for the handful of situations we currently support.

Reply:
Awesome